static/js/syntaxModesRules.ts

```diff
@@ -437,48 +457,45 @@ function rule_is(next?: string) {
 /**
  * Add a 'print' rule with brackets
  */
-function rule_printParen() {
+function rule_printParen(_PRINT: string) {
   return recognize('start', {
-    regex: '(print)(\\()',
+    regex: '(' + _PRINT + ')(\\()',
     token: ['keyword', 'paren.lparen'],
     next: 'start'
   });
 }
 
-function rule_turtle() {
+function rule_turtle(_TURN: string, _FORWARD: string) {
     return comp(
       recognize('start', {
-        regex: 'turn (left|right)?',
+        // Note: left and right are not yet keywords
+        regex: _TURN + ' (left|right)?',
         token: 'keyword',
         next: 'start',
       }),
       recognize('start', {
-        regex: 'forward',
+        regex: _FORWARD,
         token: 'keyword',
         next: 'start',
       })
     )
 }
 
-function rule_sleep() {
-    return recognize('start', {
-        regex: 'sleep',
-        token: 'keyword',
-        next: 'start',
-      }
-    )
+function rule_sleep(_SLEEP: string) {
+  return recognize('start', {
+      regex: _SLEEP,
+      token: 'keyword',
+      next: 'start',
+    }
+  )
 }
 
-
-
-
-
 /**
  * Add an 'is input' rule with brackets
  */
-function rule_isInputParen() {
+function rule_isInputParen(_IS: string, _INPUT: string) {
   return recognize('start', {
-    regex: '(\\w+)( is input)(\\()',
+    regex: '(\\w+)( ' + _IS + ' ' + _INPUT + ')(\\()',
```

Comment:
so here instead of taking the `_IS` and `_INPUT` as arguments, can we remove it and do something like this:

```
function rule_isInputParen() {
  return recognize('start', {
    regex: '(\\w+)( ' + currentLang._IS + ' ' + currentLang._INPUT + ')(\\()',
```

I know using global state is not typically advised but I think here it makes sense

Reply:
Yeah, that should be possible, the main reason I turned them into proper methods in the first place was because I seperated them into syntaxRules, which is now gone. I'll get on that.